packages/core/src/StateMachine.ts

```diff
@@ -252,10 +249,18 @@ export class StateMachine<
    * @param event The received event
    */
   public transition(
-    state: StateValue | State<TContext, TEvent, TResolvedTypesMeta> | undefined,
-    event: Event<TEvent> | SCXML.Event<TEvent>
+    state: StateValue | State<TContext, TEvent, TResolvedTypesMeta> = this
+      .initialState,
+    event: Event<TEvent> | SCXML.Event<TEvent>,
+    actorCtx?: ActorContext<TEvent, State<TContext, TEvent, any>>
   ): State<TContext, TEvent, TResolvedTypesMeta> {
-    return this._transition(state, event);
+    const currentState = toState(state, this);
```

Comment:
Sideways note: we should have a fast path that would avoid things like this. When `.transition` is called from the interpreter etc this is always going to be a state instance.

Reply:
In the `v5/simplify-algorithm` branch, I believe this can be avoided since it requires that `machine.transition(state)` is always called with a `State` object